### ISSUE TITLE
Bug: Spinning beachball after clicking OK in Preferences and at startup

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#63]
Mac OS X:
Open Preferences, click OK without changing anything.  We get the spinning beach ball for about 5-10 seconds before you can do anything.

**Comment by mpogue2** [comment]
Also occurs at startup time.  About 10 seconds of beachball.

**Comment by mpogue2** [comment]
OK, this problem doesn't seem to exist in 2b4d042fcd9c0b22bff44b43b68f7e1a614272f5.
I am suspecting ffe6aa9a0d0578c78d685dd733fd9f59793dde6a.  
Part of the problem is that I have a BOATLOAD of cue sheets now in my Lyrics directory.  So, any operation that grovels over them in the file system will take a long time.
Dan, over to you!  (By the way, the Lyrics stuff with scoring and the pulldown is VERY cool.)

**Comment by danlyke** [comment]
Any chance you can mash a debugger "stop" a few times to do some ad-hoc
profiling and see if you can figure out where it's spending its time? I
thought the only thing I added there was looking for ".htm", ".html"
and ".txt" files as well as ".mp3"...

Dan

**Comment by danlyke** [comment]
Yeah, I need more info. I'm not seeing extended delays here on my 7 year old Linux laptop with an SSD.

It'd make sense if this was a compute issue, with extra things in `pathStack` list, but this is like the problem is the file directory loading. Which either means we're screwed, or the Mac is doing something particularly bogus, because it should be scanning those directories anyway, just not returning as many files, and if the problem is copying 10k strings then it should be CPU bound... Which any running Mac should easily outperform a 7 year old Asus on.

**Comment by danlyke** [comment]
Two questions:

First, what's the output of `time find /.../yourmusicdir -type f -print | wc -l` on a Mac?

Second, and the first will help inform this: Is this a problem with a slow filesystem (a tree with 28k files on Linux with an SSD takes about an eight seconds to scan) or with the Qt library call that we're using?

Even on my ancient spinny disk server, a ~7k file tree takes < .9 seconds to scan, so maybe the problem is how Qt implements the filtering.

**Comment by danlyke** [comment]
Okay, in the (ancient) spinny disk server, a tree with ~.5M files takes ~30 seconds to scan. So 10 seconds of beachball should be circa 160k files.

**Comment by mpogue2** [comment]
```
maya:__squareDanceMusic mpogue$ find . -type f -print | wc -l
   21766
```
...and, the response is nearly instantaneous.
My MacBook has an SSD, so it's whatever we're doing in findFilesRecursively() that takes 15 seconds...
I wonder if it's faster to shell out to the OS with a `find . -type f -print`?

**Comment by mpogue2** [comment]
OK, I think I fixed this one: 180ba9e75e608a226d4ee4351817193842b27227
The problem was that resolving the (possible) aliases was taking most of the time.  Now, instead of resolving the aliases at findFilesRecursively() time, I resolve the alias at loadMP3File() time.  With 18K lyrics files in the lyrics/ tree, it was taking about 15 seconds.  Now it takes <1 sec.  And, double-click-to-load-a-song-time is not noticeably increased (we're only following the one possible alias now).
There is a bug created by this simple fix, which I'll open shortly.
